pyccel/codegen/printing/ccode.py

```diff
@@ -542,8 +547,21 @@ def stored_in_c_pointer(self, a):
     def _print_FunctionCall(self, expr):
         func = expr.funcdef
          # Ensure the correct syntax is used for pointers
-        args = [VariableAddress(a) if isinstance(a, Variable) and self.stored_in_c_pointer(f) else a
-            for a, f in zip(expr.arguments, func.arguments)]
+
+        args = []
+        for a, f in zip(expr.arguments, func.arguments):
+            if isinstance(a, Variable) and self.stored_in_c_pointer(f):
+                args.append(VariableAddress(a))
+            elif f.is_optional and not isinstance(a, Nil):
+                tmp_var_name = self._parser.get_new_name('tmp')
+                tmp_var = Variable(name = tmp_var_name, dtype = f.dtype)
+                self._additional_declare.append(tmp_var)
+                assign = Assign(tmp_var, a)
+                args.append(VariableAddress(tmp_var))
+                self._additional_code += self._print(assign) + '\n'
```

Comment:
Can we make a function for these lines, e.g:
```
def create_tmp_var(init_val, match_var)
```
that way we will avoid repetition in the future

Reply:
done.